src/main/java/org/spongepowered/api/world/Locatable.java

```diff
@@ -22,16 +22,13 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  * THE SOFTWARE.
  */
-package org.spongepowered.api.command.source;
-
-import org.spongepowered.api.command.CommandSource;
-import org.spongepowered.api.world.Location;
-import org.spongepowered.api.world.World;
+package org.spongepowered.api.world;
 
 /**
- * Location sources are {@link CommandSource}s that have a current location.
+ * Location sources are anything that have a current location.
  */
-public interface LocatedSource extends CommandSource {
+@FunctionalInterface
+public interface Locatable {
```

Comment:
I honestly think limiting this two `World` is not such a bad idea. The command API at the moment pretty much expects world and while that could be changed / checked, if you have a world location then you can use that to retrieve a chunk location anyways. K.I.S.S